pandas/core/series.py

```diff
@@ -897,26 +908,14 @@ def __getitem__(self, key):
                 else:
                     raise
 
-        if not key_is_scalar:
-            # avoid expensive checks if we know we have a scalar
```

Comment:
This comment might be relevant? Also if you move it up as you did now, you can keep this scalar check?